server/src/weaverbird/pipeline/steps/utils/combination.py

```diff
@@ -17,9 +16,21 @@ def __hash__(self):
         return hash(f"__ref__{self.uid}")
 
 
-PipelineOrDomainName = (
-    list[dict] | list["PipelineStep"] | str
-)  # can be either a domain name or a complete pipeline
+def _convert_basestep_to_dict(value: Any) -> Any:
+    if isinstance(value, BaseStep):
+        return value.model_dump()
+    elif isinstance(value, list) and all(isinstance(v, BaseStep) for v in value):
+        return [v.model_dump() for v in value]
+    return value
+
+
+# FIXME: pydantic v2 is stricter with typing, and requires this to be
+# list[dict] | str | PipelineStep. However it would result in a circular import.
+# In python 3.11, importing the PipelineStep only when TYPE_CHECKING is True works,
+# but seems a bit hacky. So this ensures that the input can only be a dict
```

Comment:
why not ca me derangeait pas le truc avant

Reply:
Ça pète en python 3.10

Reply:
arf